List item Meeting Update

daanishnasir opened the pull request:
1. Add prop for statusColor to change color of left status indicator.
2. Add prop includeDate boolean to adjust style of date/time info as seen in invision.   
 -  Move date on top; start time - end time under
3. Add date prop to pass in date ex: (date = 'January 28, 2015')

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -103,7 +105,14 @@ class ListItemMeeting extends React.PureComponent {
       if (isAllDay) {
         return <span>All day</span>;
-      } else if (time.start) {
+      }else if(includeDate && time.start) {
```
Spaces

daanishnasir replied:
done

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -73,18 +72,21 @@ class ListItemMeeting extends React.PureComponent {
       anchorOnClick,
       childrenRight,
       className,
+      date,
```
Any reason we didn't give them the timeNode option for passing in a node?

daanishnasir replied:
added timeNode option

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -103,7 +105,14 @@ class ListItemMeeting extends React.PureComponent {
+          <span key='date'>{date}</span>,
+          time.end ? <span key='time'>{time.start} - {time.end}</span> : <span key='time-0'>{time.start}</span>
+        ];
+      }else if(includeDate && date){
```
Spaces

daanishnasir replied:
done

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -103,7 +105,14 @@ class ListItemMeeting extends React.PureComponent {
+        ];
+      }else if(includeDate && date){
+        return [<span key='date'>{date}</span>];
+      }else if (time.start) {
```
Spaces

daanishnasir replied:
done

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -103,7 +105,14 @@ class ListItemMeeting extends React.PureComponent {
+      }else if(includeDate && time.start) {
+        return [
+          <span key='date'>{date}</span>,
+          time.end ? <span key='time'>{time.start} - {time.end}</span> : <span key='time-0'>{time.start}</span>
```
Instead of making this a long ternary in an array - we can wrap the logic in the span itself:
```
<span key='time'>{time.start + `${time.end ? ` - ${time.end}` : ''}`}</span>
```

daanishnasir replied:
replaced

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -103,7 +105,14 @@ class ListItemMeeting extends React.PureComponent {
+          time.end ? <span key='time'>{time.start} - {time.end}</span> : <span key='time-0'>{time.start}</span>
+        ];
+      }else if(includeDate && date){
+        return [<span key='date'>{date}</span>];
```
No need to wrap this in an array with a key if it's the only return just like isAllDay

daanishnasir replied:
Fixed

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -112,9 +121,12 @@ class ListItemMeeting extends React.PureComponent {
-        {inProgress && <span className='cui-list-item-meeting__progress-line'/>}
-        {getTime()}
+      <ListItemSection
+        key="child-0"
```
Make all single quotes

daanishnasir replied:
replaced all double to single quotes

bfbiggs commented on src/lib/ListItemMeeting/index.js:
```diff
@@ -112,9 +121,12 @@ class ListItemMeeting extends React.PureComponent {
+        key="child-0"
+        position="left"
+        includeDate={includeDate}>
+          {inProgress && <span style={{backgroundColor:statusColor}} className='cui-list-item-meeting__progress-line'/>}
```
This line is on the long side now. Maybe copy anchorLabel format below.

daanishnasir replied:
changed format to multi line like anchorLabel

bfbiggs commented on src/lib/ListItemSection/index.js:
```diff
@@ -7,31 +7,38 @@
+    const {
+      children,
+      className,
+      includeDate,
```
ListItemSection is meant to remain a very generic component.  I think you might want to include a className for what you're trying to accomplish here.

daanishnasir replied:
moved inline style width to classname

daanishnasir replied:
Removed changes in ListItemSection

bfbiggs commented:
The left spacing doesn't seem to match the design.

![screen shot 2018-08-21 at 12 23 05 pm](https://user-images.githubusercontent.com/36969382/44418913-5311d380-a53f-11e8-8767-dbd2ab4fde2f.png)

bfbiggs commented:
We'd also want to add an example to the docs as to how it is used in the design.  This helps both the implementer know how to accomplish it and helps us verify that our changes satisfy the requirements of the design.

daanishnasir commented:
![screen shot 2018-08-23 at 9 54 58 am](https://user-images.githubusercontent.com/15917537/44533152-7f059400-a6ba-11e8-83d3-f51e7b04c6a3.png)

daanishnasir commented:
![screen shot 2018-08-23 at 3 54 31 pm](https://user-images.githubusercontent.com/15917537/44551380-9cebec80-a6ec-11e8-89ad-acb98a5e77c2.png)

changed to where the user can pass in there own margins, added a dateColor prop to change color of date.
